nomt/src/commit/worker.rs

```diff
@@ -512,10 +537,18 @@ impl<H: NodeHasher> RangeCommitter<H> {
             while seeker.has_room() && start_index + pushes < self.range_end {
                 let next_push = start_index + pushes;
                 pushes += 1;
-                seeker.push(self.shared.read_write[next_push].0);
-                let blocked = seeker.submit_all(self.write_pass.downgrade())?;
-                if blocked {
-                    break;
+
+                if let Some(result) = warm_ups.remove(&self.shared.read_write[next_push].0) {
+                    warmed_up.push_back(result);
+                    if warmed_up.len() >= 512 {
```

Comment:
Is this `warmed_up` size limit there to allow the loop to continue polling on the pending paths in the seeker? to avoid storing a lot of already warmed-up paths without advancing in the seeker?

Is 512 arbitrary, or did you conduct some tests to check which number was giving enough space to the seeker without making the collection of 'warmed_up' path too slow?

Reply:
Sorry, missed this until now. It's totally arbitrary. Yes, that's the exact purpose of it.